### DISCUSSION TITLE
v2.1.0: SCS box cone API

### DISCUSSION THREAD

**Discussion by stephane-caron** [qpsolvers/qpsolvers#72] [Announcements]
This release adds support for the [SCS box cone API](https://www.cvxgrp.org/scs/api/cones.html).

Thanks to @bodono for his feedback in https://github.com/bodono/scs-python/issues/63.

### Added

- Document how to add a new QP solver to the library
- Example with (box) lower and upper bounds
- Test case where `lb` XOR `ub` is set

### Changed

- SCS: use the box cone API when lower/upper bounds are set

<hr /><em>This discussion was created from the release <a href='https://github.com/stephane-caron/qpsolvers/releases/tag/v2.1.0'>v2.1.0</a>.</em>

**Comment by bodono** [comment]
Nice work. By the way, several other solvers support box constraints (not the box cone typically, just simple upper and lower bounds on the variables).

**Reply by stephane-caron** [reply]
Absolutely :+1: I've created https://github.com/stephane-caron/qpsolvers/issues/76 to keep track.
